core/web/job_runs_controller_test.go

```diff
@@ -136,6 +136,73 @@ func TestJobRunsController_Create_Success(t *testing.T) {
 	assert.Equal(t, "100", val)
 }
 
+func TestJobRunsController_Create_Wrong_ExternalInitiator(t *testing.T) {
+	t.Parallel()
+	app, cleanup := cltest.NewApplication(t)
+	app.Start()
+	defer cleanup()
+
+	eir := &models.ExternalInitiatorRequest{
+		Name: "bitcoin",
+		URL:  cltest.WebURL(t, "http://localhost:8888"),
+	}
+	eia := models.NewExternalInitiatorAuthentication()
+	ei, err := models.NewExternalInitiator(eia, eir)
+	require.NoError(t, err)
+	assert.NoError(t, app.Store.CreateExternalInitiator(ei))
+
+	j := cltest.NewJobWithExternalInitiator(ei)
+	assert.NoError(t, app.Store.CreateJob(&j))
+
+	wrongEIR := &models.ExternalInitiatorRequest{
+		Name: "bitcoin",
+		URL:  cltest.WebURL(t, "http://localhost:8888"),
+	}
+	wrongEIA := models.NewExternalInitiatorAuthentication()
+	wrongEI, err := models.NewExternalInitiator(wrongEIA, wrongEIR)
+	require.NoError(t, err)
+	assert.NoError(t, app.Store.CreateExternalInitiator(wrongEI))
+
+	// Set up AUTH
+	headers := make(map[string]string)
+	headers[web.ExternalInitiatorAccessKeyHeader] = wrongEIA.AccessKey
+	headers[web.ExternalInitiatorSecretHeader] = wrongEIA.Secret
+
+	url := app.Config.ClientNodeURL() + "/v2/specs/" + j.ID.String() + "/runs"
+	bodyBuf := bytes.NewBufferString(`{"result":"100"}`)
+	resp, cleanup := cltest.UnauthenticatedPost(t, url, bodyBuf, headers)
+	defer cleanup()
+	assert.Equal(t, http.StatusUnauthorized, resp.StatusCode, "Response should be forbidden")
```

Comment:
Should the routes be covered in these tests?  The test logic may be reduced if we minimize the scope of the test.